### PR TITLE
fix: remove conflicting validation to fix e2e on friday evening (CMC-1104)

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
[CMC-1104](https://tools.hmcts.net/jira/browse/CMC-1104)

### Change description ###
This PR removes the conflicting validation from `confirmServiceDate` peoperty of CaseData. 

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/unspec/model/CaseData.java [hunk 1]
```diff
@@ -30,7 +29,6 @@
 
 @Data
 @Builder(toBuilder = true)
-@HasDeemedDateOfServiceTheSameAsOrAfterIssueDate(groups = ConfirmServiceDateGroup.class)
```
Comment (pd-dudkiewicz) [inline src/main/java/uk/gov/hmcts/reform/unspec/model/CaseData.java:33]: Can we remove all related code as a part of this ticket? Looks like, annotation, validation group, validator, and tests will be unnecessaary.

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/reform/unspec/model/CaseData.java:33]: done

File: e2e/api/steps.js [hunk 2]
```diff
@@ -55,8 +55,6 @@ module.exports = {
 
     await assertCallbackError('ServedDocuments', data[eventName].invalid.ServedDocuments.blankOtherDocuments,
       'CONTENT TBC: please enter a valid value for other documents');
-    await assertCallbackError('Date', data[eventName].invalid.Date.threeDaysBeforeToday,
```
Comment (pd-dudkiewicz) [inline e2e/api/steps.js:58]: `threeDaysBeforeToday` entry could be removed from api test fixture

**Review by pd-dudkiewicz — DISMISSED** [review]

File: e2e/api/steps.js (hunk 2 above)
Comment (sabahirfan) [reply e2e/api/steps.js:58]: removed

**Review by pd-dudkiewicz — APPROVED** [review]

**Review by swalker125 — APPROVED** [review]

**Pull request merged**